### ISSUE TITLE
Copy/paste <img> elements should use relative URLs

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#162]
ORIGINAL REPORT:

Right now, if you drag/drop a raster from the Files palette, the URL property
in the widget-specific section of Properties palette gets an absolute URL. For
example, when running local install, I get URLs such as
http://localhost:55555/davinci/user/davinciUser/ws/workspace/img/masthead_blue_gradient2.png.
Instead, we need to create a relative URL (e.g.,
img/masthead_blue_gradient2.png). With absolute URLs, the URL won't work if the
HTML file is transferred from one workspace to another.

UPDATED REPORT:

We seem to be using relative URLs now when dragging/dropping images from the Files palette and dragging the IMG element from the widget palette and picking a URL from the file dialog.

What's broken is copy/paste:
- New HTML File
- Go into split view
- Drag/drop SampleBanner.png from Files palette onto canvas

=> Notice that the URL on the <img> element is `src="SampleBanner.png"`
- Using context menu, copy, then paste

=> Result: the URL on the pasted <img> element is `src="http://localhost:8081/maqetta/user/maqettaUser/ws/workspace/project1/SampleBanner.jpg"`

**Comment by peller** [comment]
note that the url in the source is a relative url

**Comment by JonFerraiolo** [comment]
Moved to M4. I've run into this issue myself when using Maqetta. It's common that people will include <img> elements and common that HTML files will be transferred to a different system or user workspace.

**Comment by peller** [comment]
so the HTML source seems to do the right thing, it's just the properties palette that's broken?

**Comment by JonFerraiolo** [comment]
After further investigation, updated bug title and bug report, and lowered priority to low.

**Comment by peller** [comment]
Ok, but note also that even for the drag/drop case, the properties palette has the wrong (absolute) url.  Don't know if this is related or a separate bug.

**Comment by JonFerraiolo** [comment]
Yes, you are right, properties palette is showing the wrong value. Both problems are probably because the browser gives the absolute value when doing getAttribute('src'). In the case, perhaps the model comes to the rescue? I would guess the model is storing relative values because that's what we see in source view.

**Comment by peller** [comment]
getAttribute from the browser returns the url as-is, not absolute unless that's the way it's specified in the markup.  Perhaps it's the other way around, that our model has an absolute url?

**Comment by JonFerraiolo** [comment]
Ah, it's the 'src' property on HTMLImageElement that holds an absolute URL. Therefore, we should use `getAttribute()` instead of ``elem.src`.

```
<!DOCTYPE html>
<script type='text/javascript'>
window.onload = function(){
    var img = document.getElementById('foo');
    alert('img.src='+img.src+',img.getAttribute("src")='+img.getAttribute("src"));
}
</script>
<img id='foo' src='thumb-ibm.gif'></img>
```

shows: `img.src=file:///Users/jonferraiolo/play/thumb-ibm.gif,img.getAttribute("src")=thumb-ibm.gif`

**Comment by peller** [comment]
good catch!

**Comment by JonFerraiolo** [comment]
Pushing to M5, Priority-Med

**Comment by jhpedemonte** [comment]
-> M6

**Comment by JonFerraiolo** [comment]
-> M6/High - this causes ZIP downloads to not work without some hand-editing of URLs

**Comment by childsb** [comment]
@JonFerraiolo @peller Is this fixed?  I've tried the following scenarios without error:

1) create folder1
2) upload a jpg file to folder1
3) create a new HTML in the root
4) DND jpg in folder1 to new html
5) notice that the prop palette has proper URL in widget specific properties
6) copy then paste the image.  notice the URL in the widget specific properties is relative/correct.

and i've tried the above with image in root folder.  i'm thinking this was fixed somewhere else?
